streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -158,6 +162,12 @@ public RocksDBCacheEntry(V value, boolean isDirty) {
 
     @SuppressWarnings("unchecked")
     public void openDB(ProcessorContext context) {
+        final Map<String, Object> configs = context.appConfigs();
+        final Class<RocksDBConfigSetter> configSetterClass = (Class<RocksDBConfigSetter>) configs.get(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG);
+        if (configSetterClass != null) {
+            final RocksDBConfigSetter configSetter = Utils.newInstance(configSetterClass);
+            configSetter.setConfig(name, options);
```

Comment:
if you think theres something useful in `configs` for a `configSetter` to use, we could pass the `configs` in here as well so a `configSetter` wouldn't have to go get that data from elsewhere.